src/configuration/json-configuration-source.ts

```diff
@@ -1,9 +1,9 @@
 import { readJsonFile } from "../utils.js";
 import { PartialConfiguration } from "./configuration.js";
 
-export async function getJsonConfiguration(): Promise<PartialConfiguration> {
+export async function getJsonConfiguration(configFile = "settings.json"): Promise<PartialConfiguration> {

```

Comment:
I understand that the default value is set here, because the values from `default-configuration-source.ts` should have lower priority, and should thus be loaded later. I would still like it to additionally be set in the default config as well so that the `configFile` field on the `Configuration` type is also set if no value is provided; currently it is only set if it explicitly overrides the default.

This is useful if later parts of the script want to access the value. For example, the default value of `configFile` is currently not logged to log file.